src/shared/components/HelperText/HelperText.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react'
+import { HelperTextsWrapper, StyledHelperText, HelperTextCount } from './HelperText.style'
+
+export const getVariant = (warning?: boolean, error?: boolean) => {
+  if (error) {
+    return 'error'
+  }
+  if (warning && !error) {
+    return 'warning'
+  }
+}
+
+export type HelperTextProps = {
+  error?: boolean
+  warning?: boolean
+  maxLength?: number
+  helperText?: string
+  charactersCount?: number
+}
+
+const HelperText: React.FC<HelperTextProps> = ({
+  maxLength,
+  warning,
+  error,
+  helperText = '\u00A0',
+  charactersCount,
+}) => {
+  const [charactersWarning, setCharactersWarning] = useState<'warning' | 'error' | null>(null)
+
+  useEffect(() => {
+    if (!maxLength || !charactersCount) {
+      return
+    }
+    const warningLength = maxLength * 0.8
+
+    if (charactersCount > warningLength) {
+      setCharactersWarning('warning')
+    } else {
+      setCharactersWarning(null)
+    }
+    if (charactersCount > maxLength) {
+      setCharactersWarning('error')
+    }
+  }, [charactersCount, maxLength])
+
+  return (
+    <HelperTextsWrapper>
+      {/* caption cannot be child of <div> */}
+      <StyledHelperText variant="caption" as="span" helperTextVariant={getVariant(warning, error)}>
```

Comment:
I think the correct solution here is to not use `<caption>` as an HTML element when displaying the `caption` style. It should only be used within `<table>` so choosing it was probably a mistake on my part. Let's just make `<Text variant="caption">` render as `span` be default